src/charm.py

```diff
@@ -150,6 +141,19 @@ def _on_start_simulation_action(self, event: ActionEvent) -> None:
         except ChangeError as e:
             event.fail(message=f"Failed to execute simulation: {e.err}")
 
+    def _network_attachment_definition_from_config(self) -> dict[str, Any]:
+        ran_nad_config = {
+            "cniVersion": "0.3.1",
+            "type": "macvlan",
+            "ipam": {"type": "static"},
+        }
+        if (gnb_interface := self._get_gnb_interface_from_config()) is not None:
+            ran_nad_config.update({"type": "macvlan", "master": gnb_interface})
+        else:
+            ran_nad_config.update({"type": "bridge", "bridge": "ran-br"})
```

Comment:
Isn't it a bit confusing that the interface is `gnb_interface` but the bridge is `ran-br`
wouldn't using `gnbsim` everywhere be easier?

Just nitpicking.

Reply:
The name of the bridge needs to be the same on the router also. We could change the name of the interface here to be `ran_interface` instead, but it might be confusing because this is really the interface towards the `UPF`.

It is already set to `ran-br` and merged in the router, but we could also create a quick PR there to change the name. Let me know what you think would be less confusing.